src/views/UserEdit/UserEdit.js

```diff
@@ -154,6 +154,15 @@ class UserEdit extends React.Component {
       .catch((e) => showErrorCallout(e, this.context.sendCallout));
   }
 
+  deleteEmptyFields = (user) => {
+    if (!user.username) {
+      delete user.username;
+    }
+    if (!user.externalSystemId?.trim()) {
+      delete user.externalSystemId;
+    }
+  };
+
```

Comment:
you may further refine it by 
1. creating an array of user properties to be deleted
2. loop through the array
as the action is same always - delete the property, if empty.

Reply:
I would expect the "emptiness" check for these fields to be the same, or for there to be a comment explaining why they are different. 

This impl is easy, but I don't love functions with side-effects, i.e. that modify their arguments. IOW, technically it's OK but philosophically it's sketchy. If you're decide to go with because it's just so so simple (and that is a merit here), please document the side-effect.